src/torchmetrics/functional/nominal/pearson.py

```diff
@@ -0,0 +1,165 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import itertools
+from typing import Optional, Union
+
+import torch
+from torch import Tensor
+from typing_extensions import Literal
+
+from torchmetrics.functional.classification.confusion_matrix import _multiclass_confusion_matrix_update
+from torchmetrics.functional.nominal.utils import (
+    _compute_chi_squared,
+    _drop_empty_rows_and_cols,
+    _handle_nan_in_data,
+    _nominal_input_validation,
+)
+
+
+def _pearsons_contingency_coefficient_update(
+    preds: Tensor,
+    target: Tensor,
+    num_classes: int,
+    nan_strategy: Literal["replace", "drop"] = "replace",
+    nan_replace_value: Optional[Union[int, float]] = 0.0,
+) -> Tensor:
+    """Computes the bins to update the confusion matrix with for Pearson's Contingency Coefficient calculation.
+
+    Args:
+        preds: 1D or 2D tensor of categorical (nominal) data
+        target: 1D or 2D tensor of categorical (nominal) data
+        num_classes: Integer specifing the number of classes
+        nan_strategy: Indication of whether to replace or drop ``NaN`` values
+        nan_replace_value: Value to replace ``NaN`s when ``nan_strategy = 'replace```
+
+    Returns:
+        Non-reduced confusion matrix
+    """
+    preds = preds.argmax(1) if preds.ndim == 2 else preds
+    target = target.argmax(1) if target.ndim == 2 else target
+    preds, target = _handle_nan_in_data(preds, target, nan_strategy, nan_replace_value)
+    return _multiclass_confusion_matrix_update(preds, target, num_classes)
+
+
+def _pearsons_contingency_coefficient_compute(confmat: Tensor) -> Tensor:
+    """Compute Pearson's Contingency Coefficient based on a pre-computed confusion matrix.
+
+    Args:
+        confmat: Confusion matrix for observed data
+
+    Returns:
+        Pearson's Contingency Coefficient
+    """
+    confmat = _drop_empty_rows_and_cols(confmat)
+    cm_sum = confmat.sum()
+    chi_squared = _compute_chi_squared(confmat, bias_correction=False)
+    phi_squared = chi_squared / cm_sum
+
+    tschuprows_t_value = torch.sqrt(phi_squared / (1 + phi_squared))
+    return tschuprows_t_value.clamp(0.0, 1.0)
+
+
+def pearsons_contingency_coefficient(
+    preds: Tensor,
+    target: Tensor,
+    nan_strategy: Literal["replace", "drop"] = "replace",
+    nan_replace_value: Optional[Union[int, float]] = 0.0,
+) -> Tensor:
+    r"""Compute `Pearson's Contingency Coefficient`_  measuring the association between two categorical (nominal)
+    ata series.
+
+    .. math::
+        Pearson = \sqrt{\frac{\chi^2 / n}{1 + \chi^2 / n}}
+
+    where
+
+    .. math::
+        \chi^2 = \sum_{i,j} \ frac{\left(n_{ij} - \frac{n_{i.} n_{.j}}{n}\right)^2}{\frac{n_{i.} n_{.j}}{n}}
+
+    Pearson's Contingency Coefficient is a symmetric coefficient, i.e.
+
+    .. math::
+        Pearson(preds, target) = Pearson(target, preds)
+
+    The output values lies in [0, 1].
```

Comment:
not sure if this is true?
```suggestion
    The output values lies in [0, 1], with the optimal value being 0.
```